backend/blueprints/api.py

```diff
@@ -191,5 +191,22 @@ def api_v1_get_rank(id_):
     return jsonify(get_rank(id_))
 
 
+@bp.route('/playergames')
+@key_required
+@with_session
+def api_v1_get_playergames_by_rank(session=None):
+    if 'rank' in request.args:
+        rank = request.args['rank']
+    else:
+        return jsonify({'error': 'No rank supplied'}), 401
+    games = session.query(PlayerGame).filter(PlayerGame.rank == int(rank)).order_by(func.random())[:1000]
```

Comment:
You should limit the query to 1000 before it gets the result

Reply:
That's not how an ORM works.